runestone/usageAssignment/__init__.py

```diff
@@ -46,19 +46,23 @@ def __init__(self,content):
 # in html, self is sphinx.writers.html.SmartyPantsHTMLTranslator
 # The node that is passed as a parameter is an instance of our node class.
 def visit_ua_node(self,node):
-    course_name = node.ua_content['course_name']
-    s = ""
-    for d in node.ua_content['chapter_data']:
-        ch_name, sub_chs = d['ch'], d['sub_chs']
-        s += '<div class="panel-heading">'
-        s += ch_name
-        s += '<ul class="list-group">'
-        for sub_ch_name in sub_chs:
-            s += '<li class="simple">'
-            s += '<a href = "/runestone/static/%s/%s/%s.html">%s</a>' % (course_name, ch_name, sub_ch_name, sub_ch_name)
-            s += '</li>'
-        s += '</ul>'
-        s += '</div>'
+    try:
+        course_name = node.ua_content['course_name']
+
+        s = ""
+        for d in node.ua_content['chapter_data']:
```

Comment:
This seems like a big try block.  Should the check be to see ``if 'chapter_data' is in  node.ua_content:`` 

The message says that this is database related, and I'm assuming that some previous call to a database would have populated the chapter_data key of ua_content?